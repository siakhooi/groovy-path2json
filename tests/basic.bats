setup() {
    load 'common-setup'
    common_setup

    outputExpected=$BATS_TEST_FILENAME.$BATS_TEST_DESCRIPTION.expected.json
}

@test "basic" {
    tempDir=$(mktemp -d)
    cd "$tempDir"
    run path2json

    sed "s:TEMPDIR:${tempDir}:" "$outputExpected" | assert_output
}
@test "subdirectories" {
    tempDir=$(mktemp -d)
    cd "$tempDir"
    mkdir -p a/b/c
    touch a/b/c/d
    touch a/b/e
    run path2json

    sed "s:TEMPDIR:${tempDir}:g" "$outputExpected" | assert_output
}
