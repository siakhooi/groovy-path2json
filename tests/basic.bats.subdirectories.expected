{"isDirectory":true,"path":"TEMPDIR","children":[{"isDirectory":true,"path":"TEMPDIR/a","children":[{"isDirectory":true,"path":"TEMPDIR/a/b","children":[{"isDirectory":false,"path":"TEMPDIR/a/b/e","children":[]},{"isDirectory":true,"path":"TEMPDIR/a/b/c","children":[{"isDirectory":false,"path":"TEMPDIR/a/b/c/d","children":[]}]}]}]}]}